kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -51,9 +55,9 @@
  */
 public class SharedInformerFactory extends BaseOperation {
   private static final Logger log = LoggerFactory.getLogger(SharedInformerFactory.class);
-  private final Map<String, SharedIndexInformer> informers = new HashMap<>();
+  private final List<Map.Entry<OperationContext, SharedIndexInformer>> informers = new ArrayList<>();
 
-  private final Map<String, Future> startedInformers = new HashMap<>();
+  private final List<Map.Entry<OperationContext, SharedIndexInformer>> startedInformers = new ArrayList<>();
```

Comment:
I think you'll need to leave the start logic as async.  The reason to track statedInformers is to cancel the futures - I think stop should acutally stop everything, rather than only those that have been started async.